components/TransactionHistory.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react'
+import { Table } from 'react-bulma-components'
+import prepareUrl from '../util/prepareUrl'
+import fetch from 'isomorphic-unfetch'
+
+class TransactionHistory extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            transactions: [],
+        }
+    }
+
+    componentDidMount() {
+        this.updateTransactions(this.props.user_id, this.props.limit)
+    }
+
+    componentWillReceiveProps(props) {
```

Comment:
Don't refresh the transactions inside of the component. Instead load them in the parent and pass them using the props. Thus this component will only be responsible for rendering the transactions.